paimon-core/src/main/java/org/apache/paimon/operation/TagDeletion.java

```diff
@@ -88,15 +91,21 @@ public void cleanUnusedManifests(Snapshot taggedSnapshot, Set<String> skippingSe
         cleanUnusedManifests(taggedSnapshot, skippingSet, false);
     }
 
-    public Predicate<ManifestEntry> dataFileSkipper(Snapshot fromSnapshot) throws Exception {
-        return dataFileSkipper(Collections.singletonList(fromSnapshot));
-    }
+    protected Collection<ManifestEntry> readMergedDataFiles(Snapshot snapshot) throws IOException {
```

Comment:
Duplicate codes?

Reply:
The latest commit is reverted, please take a look again when you are free.